vignettes/NSSP_nowcast.Rmd

```diff
@@ -0,0 +1,510 @@
+Now that we have a dataframe of probabilistic nowcast draws for all reference times in our original reporting triangle, we can summarise the median and 50th and 95th quantiles.
+```{r}
+nowcast_summary_df <-
+  nowcast_draws_df |>
+  group_by(time) |>
+  summarise(
+    median = median(pred_count),
+    q50th_lb = quantile(pred_count, 0.25),
+    q50th_ub = quantile(pred_count, 0.75),
+    q95th_lb = quantile(pred_count, 0.025),
+    q95th_ub = quantile(pred_count, 0.975)
+  )
+```
+
+Next, we will join the initially reported data and the final evaluation data, both summarised by reference time.
+```{r}
+training_df_by_ref_date <- training_df |>
+  filter(report_date <= nowcast_date) |>
+  group_by(reference_date) |>
+  summarise(initial_count = sum(count)) |>
+  mutate(time = row_number())
+
+eval_data <- long_df |>
+  filter(
+    delay <= max_delay,
+    reference_date <= nowcast_date
+  ) |>
+  group_by(reference_date) |>
+  summarise(final_count = sum(count)) |>
+  mutate(time = row_number()) |>
+  select(time, final_count)
+
+nowcast_w_data <- nowcast_summary_df |>
+  left_join(training_df_by_ref_date,
+    by = "time"
+  ) |>
+  left_join(eval_data,
+    by = "time"
+  )
+head(nowcast_w_data)
+```
+
+# Plot against later observed "final" data
+Next we will make a plot of the nowcast
+```{r}
+combined_data <- nowcast_w_data |>
+  select(reference_date, initial_count, final_count) |>
+  distinct() |>
+  pivot_longer(
+    cols = c(initial_count, final_count),
+    names_to = "type",
+    values_to = "count"
+  ) |>
+  mutate(type = case_when(
+    type == "initial_count" ~ "Initially observed data",
+    type == "final_count" ~ "Final observed data"
+  )) |>
+  filter(reference_date >= nowcast_date - days(60))
+```
+<details><summary>Click to expand code to create the plot of the probabilistic nowcast</summary>
+```{r}
+nowcast_data_recent <- nowcast_w_data |>
+  filter(reference_date >= nowcast_date - days(60))
+
+plot_prob_nowcast <- ggplot(nowcast_data_recent) +
+  geom_line(
+    aes(
+      x = reference_date, y = median
+    ),
+    color = "gray"
+  ) +
+  geom_ribbon(
+    aes(
+      x = reference_date,
+      ymin = q50th_lb, ymax = q50th_ub
+    ),
+    alpha = 0.5,
+    fill = "gray"
+  ) +
+  geom_ribbon(
+    aes(
+      x = reference_date,
+      ymin = q95th_lb, ymax = q95th_ub
+    ),
+    alpha = 0.5,
+    fill = "gray"
+  ) +
+  # Add observed data and final data once
+  geom_line(
+    data = combined_data,
+    aes(
+      x = reference_date,
+      y = count,
+      color = type
+    )
+  ) +
+  theme_bw() +
+  scale_color_manual(
+    values = c(
+      "Initially observed data" = "darkred",
+      "Final observed data" = "black"
+    ),
+    name = ""
+  ) +
+  xlab("Date of ED visit") +
+  ylab("Number of ARI cases") +
+  theme(legend.position = "bottom") +
+  ggtitle("Comparison of cases of ARI as of the nowcast date, later observed,\n and generated as a probabilistic nowcast") # nolint
+```
+</details>
+```{r}
```

Comment:
Soo... is this good am I done? what do I do now? 

needs some kind of comment some kind of summary links to more resourcesx